src/components/FlatList/__tests__/index-test.js

```diff
@@ -0,0 +1,167 @@
+/* eslint-env jasmine, jest */
+
+import FlatList from '..';
+import VirtualizedList from '../../VirtualizedList';
+import View from '../../View';
+import React from 'react';
+import { mount, render } from 'enzyme';
+
+describe('components/FlatList', () => {
+  const data = [{ name: '1' }, { name: '2' }, { name: '3' }];
+  const renderItem = ({ item, index }) => (
+    <View className="item">{item.name}</View>
+  );
+  const defaultProps = { data, renderItem };
+
+  // Props
+  describe('ItemSeparatorComponent', () => {
+    const SeparatorComponent = () => <View className="separator" />;
+
+    test('renders the item between every list item', () => {
+      const wrapper = render(
+        <FlatList
+          {...defaultProps}
+          ItemSeparatorComponent={SeparatorComponent}
+        />
+      );
+
+      expect(wrapper.find('.separator')).toHaveLength(2);
+    });
+  });
+  describe('ListFooterComponent', () => {
+    const FooterComponent = () => <View className="footer" />;
+
+    test('renders 1 footer', () => {
+      const wrapper = render(
+        <FlatList {...defaultProps} ListFooterComponent={FooterComponent} />
+      );
+
+      expect(wrapper.find('.footer')).toHaveLength(1);
+    });
+  });
+  describe('ListHeaderComponent', () => {
+    const HeaderComponent = () => <View className="header" />;
+
+    test('renders 1 header', () => {
+      const wrapper = render(
+        <FlatList {...defaultProps} ListHeaderComponent={HeaderComponent} />
+      );
+
+      expect(wrapper.find('.header')).toHaveLength(1);
+    });
+  });
+  // describe("columnWrapperStyle"); // not supported
+  // describe("numColumns"); // not supported
```

Comment:
Not a limitation at all. Just not supported right now. Can work on it in a future iteration.